pybop/_problem.py

```diff
@@ -45,6 +47,11 @@ def __init__(
         self._time_data = None
         self._target = None
 
+        if isinstance(model, (pybop.BaseModel, pybop.lithium_ion.EChemBaseModel)):
```

Comment:
this is unexpected, why is `default_variables` only used for these particular models?

Reply:
I don't really understand what a `default_variables` is either! But perhaps a docstring would fix that :)

Reply:
Thanks @martinjrobins. Good shout, I'll add a docstring. `default_variables` can be viewed as the variables needed for plotting and the inner working of the problem + cost classes. For example, we have had `Time [s]` sprinkled around the classes, inaddition to passing it from the `signal` variable occassionally. `default_variables` captures these variables without needing them added to `signal`. 

The logic can probably improve in this situation, it's currently protecting against calling pybamm specific variables (`'Time [s]'`, etc.) when using a standalone or alternative model that doesn't have those keys.

Reply:
This has been updated to `additional_variables` with a docstring added.